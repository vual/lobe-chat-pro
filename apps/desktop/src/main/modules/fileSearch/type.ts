import { FileResult, SearchOptions } from '@/types/fileSearch';

/**
 * File Search Service Implementation Abstract Class
 * Defines the interface that different platform file search implementations need to implement
 */
export abstract class FileSearchImpl {
  /**
   * Perform file search
   * @param options Search options
   * @returns Promise of search result list
   */
  abstract search(options: SearchOptions): Promise<FileResult[]>;

  /**
   * Check search service status
   * @returns Promise indicating if service is available
   */
  abstract checkSearchServiceStatus(): Promise<boolean>;

  /**
   * Update search index
   * @param path Optional specified path
   * @returns Promise indicating operation success
   */
  abstract updateSearchIndex(path?: string): Promise<boolean>;
}
