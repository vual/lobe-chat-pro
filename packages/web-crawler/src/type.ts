export interface CrawlSuccessResult {
  content?: string;
  contentType: 'text' | 'json';
  description?: string;
  length?: number;
  siteName?: string;
  title?: string;
  url: string;
}

export interface CrawlErrorResult {
  content: string;
  errorMessage: string;
  errorType: string;
  url: string;
}

export interface FilterOptions {
  // 是否启用Readability
  enableReadability?: boolean;

  pureText?: boolean;
}

type CrawlImplType = 'naive' | 'jina' | 'browserless' | 'search1api';

type CrawlImplParams<T> = T & {
  filterOptions: FilterOptions;
};

export type CrawlImpl<Params = object> = (
  url: string,
  params: CrawlImplParams<Params>,
) => Promise<CrawlSuccessResult | undefined>;

export interface CrawlUrlRule {
  // 内容过滤配置（可选）
  filterOptions?: FilterOptions;
  impls?: CrawlImplType[];
  // URL匹配模式，仅支持正则表达式
  urlPattern: string;
  // URL转换模板（可选），如果提供则进行URL转换
  urlTransform?: string;
}
