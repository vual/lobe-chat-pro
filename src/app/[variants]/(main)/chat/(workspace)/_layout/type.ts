import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  conversation: ReactNode;
  portal: ReactNode;
  topic: ReactNode;
}
