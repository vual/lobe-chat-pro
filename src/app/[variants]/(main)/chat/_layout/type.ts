import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  session: ReactNode;
}
