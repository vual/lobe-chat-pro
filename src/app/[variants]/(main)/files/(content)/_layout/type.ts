import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  modal: ReactNode;
}
