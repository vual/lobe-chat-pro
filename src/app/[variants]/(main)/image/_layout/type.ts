import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  menu: ReactNode;
  topic: ReactNode;
}
