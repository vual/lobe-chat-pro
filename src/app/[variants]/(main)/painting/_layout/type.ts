import { ReactNode } from 'react';

export interface LayoutProps {
  children: ReactNode;
  palette: ReactNode;
}
