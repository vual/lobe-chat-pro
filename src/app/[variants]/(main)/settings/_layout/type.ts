import { ReactNode } from 'react';

export interface LayoutProps {
  category: ReactNode;
  children: ReactNode;
}
