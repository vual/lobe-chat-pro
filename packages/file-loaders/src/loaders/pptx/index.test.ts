// @vitest-environment node
import path from 'node:path';
import { beforeEach, describe, expect, it } from 'vitest';

import type { FileLoaderInterface } from '../../types';
import { PptxLoader } from './index';

// Import PptxLoader

// 确保你已经在 fixtures 目录下放置了 test.pptx 文件
// 这个 PPTX 文件最好包含多个幻灯片 (slides) 以便测试
const fixturePath = (filename: string) => path.join(__dirname, `./fixtures/${filename}`);

let loader: FileLoaderInterface;

const testFile = fixturePath('test.pptx'); // Use .pptx
const nonExistentFile = fixturePath('nonexistent.pptx'); // Use .pptx

beforeEach(() => {
  loader = new PptxLoader(); // Instantiate PptxLoader
});

describe('PptxLoader', () => {
  // Describe PptxLoader
  it('should load pages correctly from a PPTX file (one page per slide)', async () => {
    const pages = await loader.loadPages(testFile);
    // PPTX 文件有多少个 slide，就应该有多少个 page
    expect(pages.length).toBeGreaterThan(1);

    // 直接对整个 pages 数组进行快照测试 (会包含 slideNumber)
    expect(pages).toMatchSnapshot();
  });

  it('should aggregate content correctly (joining slides)', async () => {
    const pages = await loader.loadPages(testFile);
    const content = await loader.aggregateContent(pages);
    // 默认聚合是以换行符连接各 slide 内容
    expect(content).toMatchSnapshot('aggregated_content');
  });

  it('should handle file read errors in loadPages', async () => {
    const pages = await loader.loadPages(nonExistentFile);
    expect(pages).toHaveLength(1); // 即使失败也返回一个包含错误信息的页面
    expect(pages[0].pageContent).toBe('');
    expect(pages[0].metadata.error).toContain('Failed to load or process PPTX file:'); // Update error message check
  });
});
