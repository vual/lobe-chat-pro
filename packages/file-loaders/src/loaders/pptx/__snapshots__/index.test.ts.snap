// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PptxLoader > should aggregate content correctly (joining slides) > aggregated_content 1`] = `
"<slide_page pageNumber="1">
Hello
Page1
</slide_page>

<slide_page pageNumber="2">
Word
Page2
</slide_page>"
`;

exports[`PptxLoader > should load pages correctly from a PPTX file (one page per slide) 1`] = `
[
  {
    "charCount": 11,
    "lineCount": 2,
    "metadata": {
      "pageCount": 2,
      "slideNumber": 1,
      "sourceFileName": "test.pptx",
    },
    "pageContent": "Hello
Page1",
  },
  {
    "charCount": 10,
    "lineCount": 2,
    "metadata": {
      "pageCount": 2,
      "slideNumber": 2,
      "sourceFileName": "test.pptx",
    },
    "pageContent": "Word
Page2",
  },
]
`;
